gdx/src/com/badlogic/gdx/graphics/g2d/TextureAtlas.java

```diff
@@ -30,170 +35,157 @@
 import com.badlogic.gdx.utils.Array;
 import com.badlogic.gdx.utils.Disposable;
 import com.badlogic.gdx.utils.GdxRuntimeException;
+import com.badlogic.gdx.utils.Null;
 import com.badlogic.gdx.utils.ObjectMap;
 import com.badlogic.gdx.utils.ObjectSet;
-import com.badlogic.gdx.utils.Sort;
 import com.badlogic.gdx.utils.StreamUtils;
 
-import java.io.BufferedReader;
-import java.io.IOException;
-import java.io.InputStreamReader;
-import java.util.Comparator;
-import java.util.HashSet;
-import java.util.Set;
-
 /** Loads images from texture atlases created by TexturePacker.<br>
  * <br>
  * A TextureAtlas must be disposed to free up the resources consumed by the backing textures.
  * @author Nathan Sweet */
 public class TextureAtlas implements Disposable {
-	static final String[] tuple = new String[4];
-
 	private final ObjectSet<Texture> textures = new ObjectSet(4);
 	private final Array<AtlasRegion> regions = new Array();
 
 	public static class TextureAtlasData {
-		public static class Page {
-			public final FileHandle textureFile;
-			public Texture texture;
-			public final float width, height;
-			public final boolean useMipMaps;
-			public final Format format;
-			public final TextureFilter minFilter;
-			public final TextureFilter magFilter;
-			public final TextureWrap uWrap;
-			public final TextureWrap vWrap;
-
-			public Page (FileHandle handle, float width, float height, boolean useMipMaps, Format format, TextureFilter minFilter,
-				TextureFilter magFilter, TextureWrap uWrap, TextureWrap vWrap) {
-				this.width = width;
-				this.height = height;
-				this.textureFile = handle;
-				this.useMipMaps = useMipMaps;
-				this.format = format;
-				this.minFilter = minFilter;
-				this.magFilter = magFilter;
-				this.uWrap = uWrap;
-				this.vWrap = vWrap;
-			}
-		}
-
-		public static class Region {
-			public Page page;
-			public int index;
-			public String name;
-			public float offsetX;
-			public float offsetY;
-			public int originalWidth;
-			public int originalHeight;
-			public boolean rotate;
-			public int degrees;
-			public int left;
-			public int top;
-			public int width;
-			public int height;
-			public boolean flip;
-			public int[] splits;
-			public int[] pads;
-		}
+		static final String[] entry = new String[5];
 
 		final Array<Page> pages = new Array();
 		final Array<Region> regions = new Array();
 
 		public TextureAtlasData (FileHandle packFile, FileHandle imagesDir, boolean flip) {
-			BufferedReader reader = new BufferedReader(new InputStreamReader(packFile.read()), 64);
+			ObjectMap<String, Field<Page>> pageFields = new ObjectMap(15, 0.99f);

```

Comment:
Minor nitpick: Both `pageFields` and `regionFields` below don't need the given capacity (15 and 127 when they are filled here with 5 and 6 items, respectively). `new ObjectMap<String, Field<Page>>(7, 0.9f);` and `new ObjectMap<String, Field<Region>>(7, 0.9f);` would work with the given String keys, and also save an unchecked warning. These only seem to be used during the constructor, and it isn't called often at all, so this doesn't really matter, it's just a code-cleanliness thing. A larger capacity makes sense if more fields are expected later.

Reply:
I chose the sizes to avoid collisions in the map. It's a micro optimization and not terribly important. I'll add a comment to make it clear, or we could use the smaller sizes you listed.

Reply:
Yeah, it doesn't really matter. I did check that `(7, 0.9f)` doesn't enlarge the map capacity, though I didn't check for collisions, and 15 and 127 are the first sizes where those String keys never collide. There's much bigger performance concerns if you're constructing a TextureAtlas often enough to notice!